### ISSUE TITLE
adds cancellation of the source future support 

### ISSUE THREAD

**Issue by He-Pin** [akka/akka#31622]
Current Source.completionStage is using `toScala` which does not support this.
refs: https://github.com/reactor/reactor-core/pull/3146
```scala
  def completionStage[T](completionStage: CompletionStage[T]): Source[T, NotUsed] =
    future(completionStage.toScala)
```

**Comment by He-Pin** [comment]
Yes, if the associated stage is cancled then try to cancel the origin `CompletionStage` just as the https://github.com/reactor/reactor-core/pull/3146, The killer switch will not call the `java.util.concurrent.CompletableFuture#cancel` ,am I missed something?

**Comment by johanandren** [comment]
You would compose and call cancel from the kill switch callback yourself, if such functionality is what you want.

**Comment by He-Pin** [comment]
But why not provides that out of box, just two lines of code.
![image](https://user-images.githubusercontent.com/501740/193801556-aa656972-d077-4f16-8ccb-52dbe64c61a0.png)
